ui/src/router.tsx

```diff
@@ -167,7 +172,8 @@ export function useBaseParams(): BaseParams {
 
 // TODO(tjennison): This is becoming spaghetti. Consider alternative ways to set
 // this up or perhaps alternative libraries.
-const prefix = "tanagra/underlays/:underlayName/studies/:studyId/";
+const underlayPrefix = "tanagra/underlays/:underlayName";
+const prefix = underlayPrefix + "/studies/:studyId/";
```

Comment:
nit: Can you use these in the paths up above?

Reply:
I just reverted that change since I wasn't actually using the split version. It's a bit of a mess since I need several different pieces in different places and I think naming the little fragments just ends up being harder to read.